### PR TITLE
add agent framework security it tests

### PR DESCRIPTION
**Pull request by Zhangxunmt**
### Description
Add Security IT tests for Agent Framework. The tests all verified by a local 2.13 cluster.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java [hunk 1]
```diff
@@ -746,6 +748,47 @@ public String registerModel(String input) throws IOException {
         return parseTaskIdFromResponse(response);
     }
 
+    public void registerMLAgent(RestClient client, String input, Consumer<Map<String, Object>> function) throws IOException {
```
Comment (dhrubo-os) [inline plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java:751]: Just curious why don't we add `@Test` with the test methods here?

**Comment by dhrubo-os** [comment]
Thanks for adding integration tests. Is it possible to include testing for hidden agent use case too?

File: plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java [hunk 2]
```diff
@@ -122,7 +122,7 @@ public RestChannelConsumer prepareRequest(RestRequest request, NodeClient client
     MLPredictionTaskRequest getRequest(String modelId, String algorithm, RestRequest request) throws IOException {
         if (FunctionName.REMOTE.name().equals(algorithm) && !mlFeatureEnabledSetting.isRemoteInferenceEnabled()) {
             throw new IllegalStateException(REMOTE_INFERENCE_DISABLED_ERR_MSG);
-        } else if (FunctionName.isDLModel(FunctionName.from(algorithm)) && !mlFeatureEnabledSetting.isLocalModelEnabled()) {
+        } else if (FunctionName.isDLModel(FunctionName.from(algorithm.toUpperCase())) && !mlFeatureEnabledSetting.isLocalModelEnabled()) {
```
Comment (ylwu-amzn) [inline plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java:125]: Should we fix in `FunctionName.from(..)` method ? So we don't need to add toUpperCase in multiple places where from method being used.

Comment (thedhrub0) [inline plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java:125]: I think we should change [here](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/FunctionName.java#L36C41-L36C46)

`return valueOf(value.toUpperCase(Locale.ROOT));`

Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java:125]: Suggest add Locale `toUpperCase(Locale.ROOT)`

Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java:125]: Updated

Comment (Zhangxunmt) [reply plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java:125]: done

File: plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java (hunk 1 above)
Comment (Zhangxunmt) [reply plugin/src/test/java/org/opensearch/ml/rest/MLCommonsRestTestCase.java:751]: not needed once you extended OpenSearchRestTestCase.

**Review by jngz-es — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java (hunk 2 above)
Comment (dhrubo-os) [reply plugin/src/main/java/org/opensearch/ml/rest/RestMLPredictionAction.java:125]: Do we need to keep this: `algorithm.toUpperCase()`?

**Review by ylwu-amzn — APPROVED** [review]

**Pull request merged**